internal/batches/executor/task.go

```diff
@@ -41,7 +41,12 @@ func (t *Task) ArchivePathToFetch() string {
 	return ""
 }
 
-func (t *Task) cacheKey(globalEnv []string) *cache.ExecutionKeyWithGlobalEnv {
+func (t *Task) cacheKey(globalEnv []string, isRemote bool) *cache.ExecutionKeyWithGlobalEnv {
+	var metadataRetriever cache.MetadataRetriever
+	// If the task is being run locally, set the metadata retrieve to use the filesystem based implementation.
+	if !isRemote {
+		metadataRetriever = cache.FileMetadataRetriever{}
```

Comment:
I think personal opinion would be to move the implementation to this repo and only have an interface in lib, but no strong feelings about it.